pyClient/zethGRPC.py

```diff
@@ -175,18 +175,18 @@ def hexadecimalDigestToBinaryString(digest):
     binary = lambda x: "".join(reversed( [i+j for i,j in zip( *[ ["{0:04b}".format(int(c,16)) for c in reversed("0"+x)][n::2] for n in [1,0]])]))
     return binary(digest)
 
-# Returns nf = sha256(1110 || [a_sk]_252 || rho)
+# Returns nf = blake2s(1110 || [a_sk]_252 || rho)
 def computeNullifier(zethNote, spendingAuthAsk):
     binaryAsk = hexadecimalDigestToBinaryString(spendingAuthAsk)
     first252Ask = binaryAsk[:252]
     leftLegBin = "1110" + first252Ask
     leftLegHex = "{0:0>4X}".format(int(leftLegBin, 2))
-    nullifier = hashlib.sha256(
+    nullifier = blake2s(
```

Comment:
The nullifier should be computed by a call to the compression function of blake2s, not the full hash, right? Don't you use the full hash here?

Reply:
> Don't you use the full hash here?
You are correct. 

Using the full hash on 2 inputs of size _bloc\_size_ is equivalent to using the compression function.